src/chapters/angular-lsn2/studio.rst

```diff
@@ -0,0 +1,265 @@
+
+.. figure:: figures/add-crew.gif
+       :alt: Animated gif of crew member being added to list after add button is clicked.
+
+       Example of crew member being added.
+
+Add this code to the *bottom* of ``src/app/crew/crew.component.html``.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <input #name type="text"/>
+   <label>First mission<input #firstMission type="checkbox"/></label>
+   <button (click)="add(name.value, firstMission.checked)">Add</button>
+
+On line 1 there is an input that declares a local variable named ``#name``. On line 2 a checkbox is defined that declares a ``#firstMission``
+variable. On line 3 there is a button that when clicked sends the value of the name input and checkbox value to a function defined in ``src/app/crew/crew.component.ts``.
+
+Add the below ``add`` function to the crew component in file ``src/app/crew/crew.component.ts``.
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   add(memberName: string, isFirst: boolean) {
+     this.crew.push({name: memberName, firstMission: isFirst});
+   }
+
+Remove Crew Members
+^^^^^^^^^^^^^^^^^^^
+Allow removing of crew members by adding a remove button next to each person in the crew list.
+When the remove button is clicked a remove function in the crew component will be called which
+will remove that person from the crew array.
+
+.. figure:: figures/remove-crew.gif
+       :alt: Animated gif of crew member disappearing from the list after the remove button for that item is clicked.
+
+       Example of crew member being removed.
+
+Add line 3 to file ``src/app/crew/crew.component.ts``. Be sure to put it before the closing ``</li>``,
+so that the button appears next to each item in the crew list.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <li *ngFor="let member of crew">
+      {{member.name}}
+      <button (click)="remove(member)">remove</button>
+   </li>
+
+Add the below ``remove`` function to the crew component in file ``src/app/crew/crew.component.ts``.
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   remove(member: object) {
+     let index = this.crew.indexOf(member);
+     this.crew.splice(index, 1);
+   }
+
+Edit Crew Members
+^^^^^^^^^^^^^^^^^
+Finally we are going to allow the user to edit crew members that have already been added.
+If the crew member name is clicked, then the crew member name should be replaced with a text
+input and a save button. This means that the crew member name can be displayed in display state
+or in edit state. Only one crew member can be edited at once.
+
+.. figure:: figures/edit-crew-name.gif
+       :alt: Animated gif of crew member name being clicked, edited, and then saved.
+
+       Example of crew member name being edited.
+
+We need to add a click event to the member name. To do that, put ``{{member.name}}`` inside of a
+``<span>`` that has a ``(click)`` handler. Make your ``<li>`` in ``src/app/crew/crew.component.html``
+look like the below code.
```

Comment:
"...code below", not "...below code."
Alternatively, you can just end with "...look like:"